daml-script/test/abort/Abort.daml

```diff
@@ -0,0 +1,159 @@
+-- Copyright (c) 2023 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+-- SPDX-License-Identifier: Apache-2.0
+
+module Abort where
+
+import DA.Exception
+import Daml.Script
+
+exception CustomError
+
+try_ok_catch_AnyException_abort : Script ()
+try_ok_catch_AnyException_abort = do
+  isThrown <-
+    try do pure False
+    catch (e : AnyException) -> pure True
+  debug isThrown
+  if isThrown
+    then pure ()
+    else abort "expected exception but didn't get one"
```

Comment:
these are all variations where the only thing that changes is the caught type and the action in the `else` branch. I'd also like to add tests where the inner action throws one of the exceptions, instead of the current one that just returns `False`, to check that when it's caught the `Script` exits successfully. However it gets painful to write them manually, really I'd rather just have a template source file filled out in Bazel, but Bazel doesn't make it easy.